cpp/include/cudf/jit/span.cuh

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2019-2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/types.hpp>
+
+namespace CUDF_EXPORT cudf {
+
+/**
+ * @brief C++20 std::span with reduced feature set.
+ *
+ */
+template <typename T>
+struct span {
```

Comment:
I suppose `cudf::device_span` cannot be used because of the `cudf::host_span` in the same header?
If so, this may be another refactor opportunity -- creating a `device_span.cuh`, etc. ?
Not something for this PR.

Reply:
yes